Use consistent terminology for parameter kinds (#1172)
dotnet/csharpstandard: standard/classes.md

##### 15.6.2.3.1 General

> Any input parameter, output parameter, or reference parameter is also a ***by-reference parameter***. A by-reference parameter is a local reference variable ([§9.7](variables.md#97-reference-variables-and-returns)); the initial referent is obtained from the corresponding argument supplied in the method invocation.

Reviewer: The addition of “also” doesn’t sound correct to me in the same way “cod and haddock are also fish” doesn’t – they *are*, not *also*, fish. But how to reword? I’ll offer:

```suggestion
Input, output, and reference parameters are ***by-reference parameter***s. A by-reference parameter is a local reference variable ([§9.7](variables.md#97-reference-variables-and-returns)); the initial referent is obtained from the corresponding argument supplied in the method invocation.
```

Feel free to wordsmith!

When the first parameter of a method includes the `this` modifier, that method is said to be an ***extension method***. Extension methods shall only be declared in non-generic, non-nested static classes. The first parameter of an extension method is restricted, as follows:

> - It may be an input parameter if the parameter has a value type

Reviewer: The rewording has lost “only” which to me links to the lead-in’s “restriction”. I'll offer:

```suggestion
- It may only be an input parameter if it has a value type
```

As usual feel free to wordsmith.

> - It may be be a reference parameter if the parameter has a value type or is a generic type constrained to struct

Reviewer: As previous but also remove the “be be” typo:

```suggestion
- It may only be a reference parameter if it has a value type or has a generic type constrained to struct
```
